### ISSUE TITLE
Code documentation

### ISSUE THREAD

**Issue by coen-hyde** [keybase/client#2044]
There are no code docs in the Go client library (not sure about the other languages). This makes it difficult for a new comer to decipher how the library works.

**Comment by cjb** [comment]
Hi, how about this?

https://keybase.io/docs/client/client_architecture

**Comment by coen-hyde** [comment]
Yeah I've read that. That's a very high level overview. I'm building something on top of the keybase client library. I'm referring to code blocks or just random code comments to explain what is going on.

**Comment by oconnor663** [comment]
The interfaces in `libkb` and most other parts of this repo are still very much up in the air. There are parts of it (like https://godoc.org/github.com/keybase/client/go/saltpack) that we've tried to make consumable, but I'm worried that if you depend on `libkb` we're going to break you frequently.

**Comment by coen-hyde** [comment]
No worries. I'm expecting things to break as things change. Code documentation is pretty hard / close to impossible to do retrospectively. I don't think you have to go hardcore and make full interface docs, just a few more code comments would be good. Apart from helping people like me building on top of Keybase it would also help people in the OS community fix bugs and new hires at Keybase onboard quicker. 

Anyway cheers

**Comment by Phil-Wheeler** [comment]
Speaking more broadly, is it possible to expand on the API documentation (particularly the login.json endpoint) to provide some examples in key languages or further guidance on how to derive and post the correct pdpka4/5 hashes? 
The existing documentation there assumes a lot of complex knowledge on the part of the developer.
